app/controllers/general.php

```diff
@@ -155,6 +156,70 @@ function router(App $utopia, Database $dbForPlatform, callable $getProjectDB, Sw
             $path .= '?' . $query;
         }
 
+        $protocol = $request->getProtocol();
+
+        // Ensure preview authorization
+        $requirePreview = \is_null($apiKey) || !$apiKey->isPreviewAuthDisabled();
+        if ($isPreview && $requirePreview) {
```

Comment:
Since this is a rather long sequence, let's add a docblock to describe the flow.